### PR TITLE
[SageMaker] Add SageMaker GraphBolt internal implementation

### PR DESCRIPTION
**Pull request by thvasilo**
*Issue #, if available:*

*Description of changes:*

* The original GraphBolt on SageMaker PR was missing the internal (GSF library) implementation, we only provided the launch script and entry point. This adds the missing module in the GraphStorm library.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/sagemaker/sagemaker_gb_convert.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    ), f"{local_dist_part_config=} does not exist."
+
+    dgl_version = importlib.metadata.version("dgl")
+    if version.parse(dgl_version) < version.parse("2.1.0"):
+        raise ValueError(
+            "GraphBolt conversion requires DGL version >= 2.1.0, "
+            f"but DGL version was {dgl_version}. "
+        )
+
+    boto_session = boto3.Session(region_name=os.environ['AWS_REGION'])
```
Comment (jalencato) [inline python/graphstorm/sagemaker/sagemaker_gb_convert.py:62]: Do we have boto3 dependencies in our docker?

File: python/graphstorm/sagemaker/sagemaker_gb_convert.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    gb_start = time.time()
+    run_gb_conversion(local_dist_part_config, njobs)
+    logging.info("GraphBolt conversion took %f sec.", time.time() - gb_start)
+
+    # Iterate through the partition data and upload only the modified/new
+    # files to the corresponding path on S3
+    upload_start = time.time()
+    for root, _, files in os.walk(os.path.dirname(local_dist_part_config)):
+        for file in files:
+            if file.endswith(("fused_csc_sampling_graph.pt", ".json")):
```
Comment (jalencato) [inline python/graphstorm/sagemaker/sagemaker_gb_convert.py:76]: How about we add a check here about if there is no ``fused_csc_sampling_graph.pt`` here? Just for consistency.

**Review by jalencato — COMMENTED** [review]
Do we have a test on SageMaker for this PR's newly introduced function?

File: python/graphstorm/sagemaker/sagemaker_gb_convert.py (hunk 1 above)
Comment (thvasilo) [reply python/graphstorm/sagemaker/sagemaker_gb_convert.py:62]: `sagemaker` depends on `boto` and `sagemaker` is installed on all SM images we use so we should be OK here.

I am planning a small update to our SM images to pin sagemaker, boto and botocore to specific versions.

**Comment by thvasilo** [comment]
> Do we have a test on SageMaker for this PR's newly introduced function?

You mean a regression test? I don't think we have any unit test for any of the sagemaker code.

**Comment by jalencato** [comment]
> > Do we have a test on SageMaker for this PR's newly introduced function?
> 
> You mean a regression test? I don't think we have any unit test for any of the sagemaker code.

I mean we add a new regression test after we merge this PR, as no current unit test covers the function here.

**Review by jalencato — APPROVED** [review]
LGTM.

**Pull request merged**